CorsixTH/Lua/research_department.lua

```diff
@@ -343,8 +343,8 @@ function ResearchDepartment:addResearchPoints(points, autopsy_room)
   ---------------------- An autopsy has been done ---------------------------
   if autopsy_room then
     -- Do something only if the room is among those not yet discovered.
-    for room, _ in pairs(self.hospital.undiscovered_rooms) do
-      if room.id == autopsy_room then
+    for room_id, room in pairs(self.hospital.room_discovery) do
+      if room_id == autopsy_room and not room.is_discovered then
```

Comment:
Why do you iterate over the map if you eventually only allow the key `autopsy_room` ? (`autopsy_room` seems a quite wrong name, but that's a different matter)